src/eins-hwinfo.c

```
@@ -135,129 +156,18 @@ eins_hwinfo_get_disk_space_for_partition (GFile   *file,
-  g_object_unref (monitor);
-}
-
-/* On the first boot, the root partition is extended to fill the disk, in the
- * background. We may be running before this process has completed; in that
- * case, we need to wait. Rather than monitoring eos-firstboot.service via
- * systemd's D-Bus API, we look for a flag file in /var.
- */
-static gboolean
-start_recording_disk_space_when_booted (gpointer data)
-{
-  g_autoptr(GFile) booted = g_file_new_for_path (BOOTED_FLAG_FILE_PATH);
-  g_autoptr(GFileMonitor) monitor = NULL;
   g_autoptr(GError) error = NULL;
 
-  if (g_file_query_exists (booted, NULL))
-    {
-      g_debug ("%s already exists", BOOTED_FLAG_FILE_PATH);
-      start_recording_disk_space ();
-    }
-  else if (!(monitor = g_file_monitor_file (booted, G_FILE_MONITOR_NONE,
-                                            NULL, &error)))
-    {
-      g_warning ("Couldn't watch %s: %s",
-                 BOOTED_FLAG_FILE_PATH, error->message);
-      start_recording_disk_space ();
-    }
-  else
-    {
-      g_debug ("Waiting for %s to appear before reporting disk space",
-               BOOTED_FLAG_FILE_PATH);
-
-      /* Ownership is transferred to boot_finished_cb() */
-      g_signal_connect (g_steal_pointer (&monitor), "changed",
-                        (GCallback) boot_finished_cb,
-                        NULL);
-    }
-
-  return G_SOURCE_REMOVE;
-}
-
-GVariant *
-eins_hwinfo_get_ram_size (void)
-{
-  glibtop_mem mem;
-  guint32 size_mib;
-
-  glibtop_get_mem (&mem);
-  size_mib = round_to_nearest (mem.total, ONE_MIB_IN_BYTES);
-  return g_variant_new_uint32 (size_mib);
-}
-
-static gboolean
-record_ram_size (gpointer unused)
-{
-  emtr_event_recorder_record_event (emtr_event_recorder_get_default (),
-                                    RAM_SIZE_EVENT,
-                                    eins_hwinfo_get_ram_size ());
-  return G_SOURCE_REMOVE;
+  eins_hwinfo_get_disk_space_for_partition (root, diskspace, &error);
```

If you are just going to ignore the error you can just pass `NULL`. But the caller should log the error with `g_warning()` as it previously did.

---

Fixed in the other way as mentioned above.